### ISSUE TITLE
Finding julia.h

### ISSUE THREAD

**Issue by barche** [JuliaLang/julia#15518]
Hi,

Is there an official way to find the `julia.h` for the running Julia instance? Currently I look in `JULIA_HOME/../include/julia`, but it seems to be different for some installations?

It would be useful to have a generic way of finding this for packages that need to build a library that needs `julia.h`.

**Comment by tkelman** [comment]
You can run `julia julia-config.jl --cflags`, but I believe that will usually return what you've already been using. Any time it's different is because someone was using a source build layout instead of a `make install` / binary layout.

**Comment by diam** [comment]
That was the point, thanks you very much!
As I didn't see any configure file which I used to use for change the prefix installation directory, I thought julia did not provide a custom installation procedure.

What I did was (since pre-Julia-0.3) :

- `make`,
- mirror all the `julia-0.4.3/usr/*` generated content to my prefix installation,
- playing with some environnement (PATH, ...) and link (current versus dev julia version).

Now I do::

- `make install`,
- mirror all the `julia-0.4.3/julia-0.4.3/*` generated content to my prefix installation.
  This added the julia.h file in the include generated directory (which was not the case
  with the previous method),
- playing with some environnement (PATH, ...) and link (current versus dev julia version).

Also I did not see the text in de README of Julia:

    ...
    write `prefix=/path/to/install/folder` into `Make.user` and then run `make install`. 
    ...

Is there a way to add this prefix directly on the command line instead tu put it in the `Make.user` file (`make --prefix=xxx install`)?

-- Maurice
